### PR TITLE
Fix updating HPO tems

### LINKED ISSUE #3430: Update HPO command crashes
After the latest HPO release, when you run `scout update hpo` you get the following error:

```
2022-06-15 11:26:03 ChiaraRMBP.local scout.build.hpo[56137] DEBUG Building hpo term HP:5000047
2022-06-15 11:26:03 ChiaraRMBP.local scout.build.hpo[56137] DEBUG Building hpo term part_of

Traceback (most recent call last):
  File "/Users/chiararasi/miniconda3/envs/py38/bin/scout", line 33, in <module>
    sys.exit(load_entry_point('scout-browser', 'console_scripts', 'scout')())
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/cli.py", line 579, in main
    return super().main(*args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/decorators.py", line 26, in new_func
    return f(get_current_context(), *args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/flask/cli.py", line 427, in decorator
    return __ctx.invoke(f, *args, **kwargs)
  File "/Users/chiararasi/miniconda3/envs/py38/lib/python3.8/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/commands/update/hpo.py", line 54, in hpo
    load_hpo_terms(adapter, hpo_lines=hpoterms, hpo_gene_lines=hpo_to_genes)
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/load/hpo.py", line 116, in load_hpo_terms
    hpo_bulk.append(build_hpo_term(hpo_info))
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/build/hpo.py", line 35, in build_hpo_term
    hpo_obj = HpoTerm(
  File "/Users/chiararasi/Documents/work/GITs/scout/scout/models/phenotype_term.py", line 30, in __init__
    self["hpo_number"] = int(hpo_id.split(":")[-1])
ValueError: invalid literal for int() with base 10: 'part_of'
(py38) chiararasi@ChiaraRMBP:~/Documents/work/GITs/scout$
```

Thanks again @leahkemp for reporting the error! I'll look into it!

### PR DESCRIPTION
**Pull request by northwestwitch**
Fix #3430 

The last lines of the hpo.obo file now look like this:

```
...

[Term]
id: HP:5000048
name: Anti-Kv1.4 antibody
def: "The presence of autoantibodies (immunoglobulins) in the blood circulation that react against the voltage-gated potassium channel Kv1.4." [PMID:22167224, PMID:30918333]
is_a: HP:0030057 ! Autoimmune antibody positivity
created_by: ORCID:0000-0002-3387-1836

[Typedef]
id: inheres_in
name: inheres in
xref: RO:0000052

[Typedef]
id: inheres_in_part_of
name: inheres in part of
xref: RO:0002314
holds_over_chain: inheres_in part_of

[Typedef]
id: part_of
name: part of
xref: BFO:0000050
```

And Scout tries to parse `[Typedef]` items as if they were genes. A solution is stopping line parsing when one reaches the first [Typedef].

<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the paxa software installed on `hasta`:
 - Log in into hasta: `ssh <USER.NAME>@hasta.scilifelab.se`
 - Activate the staging environment with the command `us` (Use Stage)
 - Run the command `paxa` and follow the instructions. Make sure to specify scout-stage as the resource you request and the server cg-vm1 as server.
1. Log out from the hasta server.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, log out from `cg-vm1` and log in again in the `hasta` server, repeat the `hasta` and `paxa` procedure, which will release the allocated resource (scout-stage) to be used for testing by other users.
</details>

**How to test**:
1. On main branch, locally, run: `scout --demo update hpo` . It will crash
2. Repeat from this branch and HPO terms should be updated

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
@lleahkemp this branch should fix the HPO terms updating thing!

File: scout/parse/hpo_terms.py [hunk 1]
```diff
@@ -48,6 +48,10 @@ def parse_hpo_obo(hpo_lines):
         if len(line) == 0:
             continue
         line = line.rstrip()
+
+        if line == "[Typedef]":
+            break
```
Comment (dnil) [inline scout/parse/hpo_terms.py:53]: Isn't this setting up for problems if the sort order were to change? How about skipping a while until a new term is found?

**Review by dnil — CHANGES_REQUESTED** [review]

**Comment by dnil** [comment]
Thank you for testing! I imagined spending a few minutes on that one!

**Review by dnil — APPROVED** [review]
this looks familiar! 😉

**Pull request merged**